packages/react/src/InlineMessage/InlineMessage.tsx

```diff
@@ -1,78 +1,104 @@
-  }
+    &[data-size='medium'] {
+      --inline-message-fontSize: var(--text-body-size-medium, ${get('fontSizes.1')});
+      --inline-message-lineHeight: var(--text-body-lineHeight-medium, 1.4285);
+    }
 
-  &[data-variant='success'] {
-    --inline-message-fgColor: ${get('colors.success.fg')};
-  }
+    &[data-variant='warning'] {
+      --inline-message-fgColor: ${get('colors.attention.fg')};
+    }
 
-  &[data-variant='unavailable'] {
-    --inline-message-fgColor: ${get('colors.fg.muted')};
-  }
+    &[data-variant='critical'] {
+      --inline-message-fgColor: ${get('colors.danger.fg')};
+    }
 
-  & .InlineMessageIcon {
-    min-height: calc(var(--inline-message-lineHeight) * var(--inline-message-fontSize));
-  }
-`
+    &[data-variant='success'] {
+      --inline-message-fgColor: ${get('colors.success.fg')};
+    }
+
+    &[data-variant='unavailable'] {
+      --inline-message-fgColor: ${get('colors.fg.muted')};
+    }
+
+    & .InlineMessageIcon {
+      min-height: calc(var(--inline-message-lineHeight) * var(--inline-message-fontSize));
+    }
+  `,
+)
 
 const variantToIcon: Record<MessageVariant, React.ReactNode> = {
-  warning: <AlertIcon className="InlineMessageIcon" />,
-  critical: <AlertIcon className="InlineMessageIcon" />,
-  success: <CheckCircleIcon className="InlineMessageIcon" />,
-  unavailable: <AlertIcon className="InlineMessageIcon" />,
+  warning: <AlertIcon className={classes.InlineMessageIcon} />,
+  critical: <AlertIcon className={classes.InlineMessageIcon} />,
+  success: <CheckCircleIcon className={classes.InlineMessageIcon} />,
+  unavailable: <AlertIcon className={classes.InlineMessageIcon} />,
 }
 
 const variantToSmallIcon: Record<MessageVariant, React.ReactNode> = {
-  warning: <AlertFillIcon className="InlineMessageIcon" size={12} />,
-  critical: <AlertFillIcon className="InlineMessageIcon" size={12} />,
-  success: <CheckCircleFillIcon className="InlineMessageIcon" size={12} />,
-  unavailable: <AlertFillIcon className="InlineMessageIcon" size={12} />,
+  warning: <AlertFillIcon className={classes.InlineMessageIcon} size={12} />,
+  critical: <AlertFillIcon className={classes.InlineMessageIcon} size={12} />,
+  success: <CheckCircleFillIcon className={classes.InlineMessageIcon} size={12} />,
+  unavailable: <AlertFillIcon className={classes.InlineMessageIcon} size={12} />,
 }
 
-export function InlineMessage({children, size = 'medium', variant, ...rest}: InlineMessageProps) {
+export function InlineMessage({children, className, size = 'medium', variant, ...rest}: InlineMessageProps) {
+  const enabled = useFeatureFlag(CSS_MODULES_FEATURE_FLAG)
+
   const icon = size === 'small' ? variantToSmallIcon[variant] : variantToIcon[variant]
+  if (enabled) {
+    return (
+      <StyledMessage
+        className={clsx(classes.InlineMessage, className)}
+        {...rest}
+        data-size={size}
+        data-variant={variant}
+      >
+        {icon}
+        {children}
+      </StyledMessage>
+    )
+  }
```

Comment:
I bet we could keep using `StyledMessage` instead of having to branch here by toggling the class name on the flag, e.g.

```tsx
    <StyledMessage {...rest} className={clsx(className, { [classes.InlineMessage]: enabled })} data-size={size} data-variant={variant}>
      {icon}
      {children}
    </StyledMessage>
```

This could help to avoid any issues later on where it's easy to forget to add something to both branches with flagged code

Reply:
That's fair, I'm still not used to reading that `{ [classes.InlineMessage]: enabled }` syntax.

Reply:
Either works to me! 👍